onnxruntime/contrib_ops/cpu/transformers/subgraph_t5_decoder.cc

```diff
@@ -238,7 +268,7 @@ Status T5DecoderSubgraph::CreateInitialFeeds(
   // When first_past_input_index_ == 3, the encoder_hidden_states and past states are copied from the second output
   // of encoder.
   // When first_past_input_index_ == 2, the past states are copied from the second output of encoder.
-  for (size_t j = static_cast<size_t>(4) - first_past_input_index_; j < encoder_fetches.size(); j++) {
+  for (size_t j = static_cast<size_t>(2) - has_hidden_state_; j < encoder_fetches.size(); j++) {
```

Comment:
The inputs `input_ids, encoder_input_ids (optional), encoder_attention_mask, encoder_hidden_states (optional), past_self_key_0, past_self_value_0, past_cross_key_0, past_cross_value_0`. 

This loop is used to add feeds for `encoder_hidden_states (optional), past_self_key_0, past_self_value_0, past_cross_key_0, past_cross_value_0`.

The encoder output is like `logits, encoder_hidden_states (optional), past_self_value_0, past_cross_key_0, past_cross_value_0, ..." so j shall starts from 2. 

Here we assume that, if encoder hidden state is not used in decoder, we shall not output it in encoder for best performance.